src/elicznik/session.py

```diff
@@ -4,6 +4,15 @@
 import requests
 
 
+def _handle_ban_and_auth(r:requests.Request, *args, **kwargs):
+    if r.url == "https://elicznik.tauron-dystrybucja.pl/blokada":
+        raise RuntimeError("you have been baned")
+
+    if r.url == "https://logowanie.tauron-dystrybucja.pl/login":
+        if 'Login lub has' in r.text:
```

Comment:
Nesting is not needed here, let's do:
```
if r.url == "https://logowanie.tauron-dystrybucja.pl/login" and 'Login lub has' in r.text:
    ...
```

Consider replacing "https://logowanie.tauron-dystrybucja.pl/login" with the constant symbol `LOGIN_URL` defined in elicznik.py.

btw why is the second condition really needed?  Isn't it sufficient to just check `r.url`?

Reply:
done,
about 'Login lub has' I don't recall the exact scenario, but the /login endpoint can return page without login form. So this is here to ensure that login page is correct